src/Sylius/Bundle/CoreBundle/Checkout/CheckoutProcessScenario.php

```diff
@@ -54,21 +48,16 @@ public function build(ProcessBuilderInterface $builder)
             ->add('payment', 'sylius_checkout_payment')
             ->add('finalize', 'sylius_checkout_finalize')
             ->add('purchase', 'sylius_checkout_purchase')
-        ;
-
-        $builder
             ->setDisplayRoute('sylius_checkout_display')
             ->setForwardRoute('sylius_checkout_forward')
             ->setRedirect('sylius_homepage')
-            ->validate(function () use ($cart) {
-                return !$cart->isEmpty();
+            ->validate(function () {
+                return !$this->cartProvider->getCart()->isEmpty();
```

Comment:
TBH. I was not sure about this, and decided to not use protected method but direct call, can change if it's a problem.